### PR TITLE
[iOS] WebView renderer that replaces UIWebView with WkWebView

### LINKED ISSUE #1666: [Enhancement] Use WKWebView on iOS
## Rationale

WKWebView has been the modern way to create webviews for some time on iOS. It would be best to support that.

## Implementation

The only backwards compat friendly mode of implementation here is to copy the old renderer to `WKWebViewRenderer` and modify it to use `WKWebView`. This renderer will then be available but not enabled by default.

### Android
No Change

### iOS
If the user puts

```csharp
[assembly: ExportRenderer(typeof(WebView), typeof(WKWebViewRenderer))]
```

in the AssemblyInfo.cs of their application, they will then use the new renderer.

### UWP
No change

## Implications for CSS
Not relevant to CSS

## Backward Compatibility

All good since this makes no modifications to existing code and is 100% opt-in.

## Difficulty : Medium

There will likely be some hiccups in the port but it looks to be fairly straightforward. There will certainly be unresolvable behavioral differences, but to a certain extent that is expected.

### PR DESCRIPTION
**Pull request by kingces95**
### Description of Change ###

Implementation of iOS WkWebViewRenderer.

WkWebViewRenderer is a port of WebViewRenderer which uses WkWebView instead of UIWebView. The port does not seek to address any shortcomings in UIWebView nor does it seek to add any new functionality provided by WkWebView.

### Issues Resolved ###

fixes #1666

### API Changes ###

None. To opt-in to using WkWebView add the following to the `AssemlyInfo.cs` file of your iOS C# project file:
``` C#
[assembly: ExportRenderer(typeof(WebView), typeof(Xamarin.Forms.Platform.iOS.WkWebViewRenderer))]
```

### Platforms Affected ###

- iOS

### Behavioral/Visual Changes ###

Hopefully minimal. Additional testing would be necessary before this renderer could be certified as a drop in replacement for the previous renderer. 

### PR Checklist ###

- [x] Has automated tests; Tests are manual but has console and VCR controls suitable to future automation
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by samhouts** [comment]
build --uitests

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+			var buttonState = new Button() { Text = "?", BackgroundColor = Color.LightBlue, AutomationId = "buttonState" };
+			var buttonStop = new Button() { Text = "X", BackgroundColor = Color.LightBlue, AutomationId = "buttonStop" };
+
+			var buttonA = new Button() { Text = "GO", BackgroundColor = Color.LightBlue, AutomationId = "buttonA" };
+			var buttonB = new Button() { Text = "HTML", BackgroundColor = Color.LightBlue, AutomationId = "buttonB" };
+			var buttonC = new Button() { Text = "EVAL", BackgroundColor = Color.LightBlue, AutomationId = "buttonC" };
+			var buttonD = new Button() { Text = "AEVAL", BackgroundColor = Color.LightBlue, AutomationId = "buttonD" };
+
+			//var url = "http://www.milliondollarhomepage.com/";
+			var url = "https://www.cnn.com/";
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: Maybe xamarin.com or something we own?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+			var buttonClear = new Button() { Text = "--", BackgroundColor = Color.LightBlue, AutomationId = "buttonClear" };
+			var buttonState = new Button() { Text = "?", BackgroundColor = Color.LightBlue, AutomationId = "buttonState" };
+			var buttonStop = new Button() { Text = "X", BackgroundColor = Color.LightBlue, AutomationId = "buttonStop" };
+
+			var buttonA = new Button() { Text = "GO", BackgroundColor = Color.LightBlue, AutomationId = "buttonA" };
+			var buttonB = new Button() { Text = "HTML", BackgroundColor = Color.LightBlue, AutomationId = "buttonB" };
+			var buttonC = new Button() { Text = "EVAL", BackgroundColor = Color.LightBlue, AutomationId = "buttonC" };
+			var buttonD = new Button() { Text = "AEVAL", BackgroundColor = Color.LightBlue, AutomationId = "buttonD" };
+
+			//var url = "http://www.milliondollarhomepage.com/";
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: don't need this commented code, right?

File: Xamarin.Forms.ControlGallery.iOS/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -40,4 +40,5 @@
 [assembly: Xamarin.Forms.ResolutionGroupName (Xamarin.Forms.Controls.Issues.Effects.ResolutionGroupName)]
 
 // Deliberately broken image source and handler so we can test handling of image loading errors
-[assembly: ExportImageSourceHandler(typeof(FailImageSource), typeof(BrokenImageSourceHandler))]
\ No newline at end of file
+[assembly: ExportImageSourceHandler(typeof(FailImageSource), typeof(BrokenImageSourceHandler))]
+[assembly: ExportRenderer(typeof(WkWebView), typeof(Xamarin.Forms.Platform.iOS.WkWebViewRenderer))]
```
Comment (samhouts) [inline Xamarin.Forms.ControlGallery.iOS/Properties/AssemblyInfo.cs]: If we're planning to support the old WebViewRenderer still, we should `#if` this so we can run tests with the old renderer, too.

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+				cancel = false;
+				navigating = false;
+			};
+
+			// Initialize ui here instead of ctor
+			Content = grid;
+			BackgroundColor = Color.Gray;
+		}
+
+		private void WebView_Navigating(object sender, WebNavigatingEventArgs e)
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: 👮 private

Comment (samhouts) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: Also, why is this here?

Comment (jassmith) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: And the underscore :/

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs (hunk 1 above)
Comment (jassmith) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: www.microsoft.com, remove all other URLs

Comment (kingces95) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: Yeah, they're fake news.

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs (hunk 2 above)
Comment (kingces95) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: Nope.

File: Xamarin.Forms.ControlGallery.iOS/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (kingces95) [reply Xamarin.Forms.ControlGallery.iOS/Properties/AssemblyInfo.cs]: Actually, this is my happy solution to that problem. `WkWebView` is only used in the 1666 unit test. All the existing tests still use `WebView` which uses the old renderer. 

I added `WkWebView` to the test in gallery. It derives from the public `WebView`. So only 1666 will use the `WkWebViewRenderer` during iOS test runs. Everything else should be the same.  

That works, no?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs (hunk 4 above)
Comment (kingces95) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: Death to `WebView_Navigating`!

**Comment by kingces95** [comment]
@samhouts Ah. Sure. Wouldn't hurt! Tho isn't that warning only issued when the compiler runs over the obsolete type referenced from another lib? Looks like for us it'd only show up in our Forwarding libraries... Customer apps don't generally point at the renderers, right?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs (hunk 1 above)
Comment (saamerm) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1666.cs]: Lol Mr. Savage

**Pull request merged**